### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by MrPetovan**
- Renamed "plugin" to "addon"
- Added ping during probe (related to #43)
- Added CLI console
- Added auto increment id field to sync-push-queue (workaround for #50)
- Reworked tag table to improve performances
- Fix bunch of PHP Notices
- Fix version check (#49)

### CODE REVIEW

File: src/Core/Console/Config.php [hunk 1]
```diff
@@ -0,0 +1,137 @@
+<?php
+
+/*
+ * To change this license header, choose License Headers in Project Properties.
```
Comment (tobiasd) [inline src/Core/Console/Config.php]: Should this be changed?

Comment (MrPetovan) [reply src/Core/Console/Config.php]: Totally, this is added by my IDE and has no business in the repo.

**Comment by AndyHee** [comment]
Can this be merged soon?

**Comment by AndyHee** [comment]
Thanks.

**Comment by MrPetovan** [comment]
You're welcome, I just found this back since I'm fixing a host of PHP 7 Warnings.

**Pull request merged**